### ISSUE TITLE
rowClick is called automatically

### ISSUE THREAD

**Issue by 7system7** [marmelab/react-admin#10100]
**What you were expecting:**
When I click on a `Datagrid`'s row, then the `rowClick` callback should be called. So, it should be called only on an `onClick` event.

**What happened instead:**
After upgrading to v5, and if I define the the `rowClick` attribute explicitly as a callback fn, then it is automatically called on `Datagrid`.

**Steps to reproduce:**
1. Install a react-admin. ` $ npx create-react-admin@latest your-admin-name`
1. Create a list w/ the attribute `rowClick` i.e.
    ```javascript
    ...
    <List>
        <Datagrid rowClick={(id, resource, record) => console.log(id)}>
    ...
    ```
1. It will be called as much as row the `Datagrid` has.

**Environment**

* React-admin version: >= 5.0.0
* Last version that did not exhibit the issue (if applicable): 4.16.20
* React version: 18.3.1
* Browser: Chromium, Firefox, Safari etc.
* Stack trace (in case of a JS error): -

**Comment by djhi** [comment]
Reproduced. That's indeed a regression

**Comment by fzaninotto** [comment]
The regression was caused by #9951

**Comment by fzaninotto** [comment]
Fixed by #10102
